[Feat] #72 - 데일리 루틴 추가 관련 API 연결 및 데일리루틴 로직 수정

frohsch opened the pull request:
## 🔥*Pull requests*

👷 **작업한 내용**
<!-- 작업한 내용을 적어주세요. -->
회원 데일리 루틴 추가 성공 API 연결 
테마별 데일리 루틴 리스트 조회 API 연결
데일리 루틴 테마 리스트 조회 API 연결
회원 데일리 루틴 삭제 API 수정
데일리루틴 뷰 로직 수정

🚨 **참고 사항**
<!-- 참고 사항을 적어주세요. 없으면 지워주세요. -->
NotificationCenter를 사용하면 반드시 !! removeObserver를 해줘야 합니다. 노티를 해제해주지 않으면 계속해서 데이터가 쌓이게 되고 분명히 1개만 눌렀는데 3개가 되는 불상사가 발생할 수 있습니다.. @leewoojye 
데일리루틴 뷰 로직은 전반적으로 수정했습니다. 달성 완료 버튼을 눌렀을 때 모든 데일리 루틴이 완료되는 것을 고쳤습니다. 제 눈에는 이제 잘 돌아가는 것 같기는 한데 QA하면서 문제점을 찾아야 할 것 같습니다. 
데일리루틴 관련해서 메서드 호출 순서가 맞지 않아서 수정했습니다. 
데일리루틴 뷰에서 이미 추가한 루틴을 추가하려고 할 때 알림창을 띄워야 할 것 같아서 이부분 기/디 쌤들께 노티 드렸고 뷰 나오면 추가하면 될 것 같습니다 !

데일리루틴뷰에서 오토레이아웃 관련 문제가 발생하는데 이부분은 새로 이슈 파서 작업하겠습니다. 


📟 **관련 이슈**
- Resolved: #72

frohsch committed:
[Feat] #72 - Implement AddDailyRoutineView API Services

frohsch committed:
[Chore] #72 - Delete unused code

frohsch committed:
[Fix] #72 - Modify Carousel pagecounts

frohsch committed:
[Merge] #72 - Commit Merge

frohsch committed:
[Fix] #72 - minor fix

frohsch committed:
[Chore] #72 - Code Cleanup

frohsch committed:
[Fix] #72 - Modi
fy DailyRoutine delete logics

frohsch committed:
[Fix] #72 - Set NotificationCenter removeObserver

frohsch committed:
[Merge] #72 - Commit Merge

frohsch committed:
[Fix] #72 - delete unused code

ahra1221 commented on Sopetit-iOS/Sopetit-iOS/Network/Base/URLConstant.swift:
```diff
@@ -33,10 +33,14 @@ enum URLConstant {
     static let patchRoutineURL = baseURL + "/api/v1/routines/daily/member/routine/"
+    static let dailyThemesURL = baseURL + "/api/v1/routines/daily/themes"
+    static let routinesDailyURL = baseURL + "/api/v1/routines/daily"
+    static let routinesDailyMemberURL = baseURL + "/api/v1/routines/daily/member"
```
p2
deleteURL 이랑 똑같은거 같은데 하나만 써주셔도 될거 같아요~

ahra1221 commented on Sopetit-iOS/Sopetit-iOS/Presentation/DailyRoutine/AddDailyRoutineScene/View/AddDailyRoutineView.swift:
```diff
@@ -69,7 +69,7 @@ final class AddDailyRoutineView: UIView {
     lazy var pageControl: UIPageControl = {
         let pageControl = UIPageControl(frame: CGRect(x: 0, y: self.frame.maxY - 100, width: self.frame.maxX, height: 50))
-        pageControl.numberOfPages = 7
+        pageControl.numberOfPages = 3
```
p3
3으로 바꾼 이유가 있나요??? 이 부분은 서버통신해서 바꿔주는거 같은데 초기화값이 바뀐 이유가 궁금함다

frohsch replied:
아 뭔가 실험해보려고 바꿔놨던거였습니다 !! 근데 이 부분 지워도 잘 작동해서 지우는게 좋을 것 같습니다!

frohsch committed:
[Merge] #72 - Commit Merge
Merge branch 'develop' of https://github.com/Team-Sopetit/Sopetit-iOS into feat/#72

ahra1221 commented on Sopetit-iOS/Sopetit-iOS/Presentation/DailyRoutine/DailyRoutineScene/View/DailyView.swift:
```diff
@@ -13,12 +13,21 @@ final class DailyView: UIView {
+//        let flowLayout = UICollectionViewFlowLayout()
+        layout.itemSize = CGSize(width: (SizeLiterals.Screen.screenWidth - 40), height: 136)
+        layout.footerReferenceSize = CGSize(width: (SizeLiterals.Screen.screenWidth - 40), height: 136)
+//        flowLayout.minimumLineSpacing = 12
+        //        flowLayout.sectionInset = UIEdgeInsets(top: 0, left: 0, bottom: 12, right: 0)
+        //        let view = UICollectionView(frame: .zero, collectionViewLayout: flowLayout)
```
p2
필요없는 주석 지워주십숑~

frohsch replied:
넹!!

ahra1221 commented on Sopetit-iOS/Sopetit-iOS/Presentation/DailyRoutine/DailyRoutineScene/ViewController/DailyViewController.swift:
```diff
@@ -227,7 +229,7 @@ extension DailyViewController: BottomSheetButtonDelegate {
         for cell in self.collectionview.visibleCells {
             if let dailyCell = cell as? DailyRoutineCollectionViewCell {
-                if dailyCell.achieveButton.isEnabled {
+                if dailyCell.index == achieveIndex {
```
p4
cell의 인덱스가 -1이 되는 부분?에 대한 코드는 어떤 로직인지 궁금합니당

frohsch replied:
그냥 초기값을 설정해뒀습니다 !! 실제로 deque 할 때 indexPath.item 값을 넣어주므로 index가 -1인 경우는 없을 것입니다!!

ahra1221 reviewed (approved):
자잘한 레이아웃 이슈들 해결과 개쩌는 api 붙이는 속도!!!! 수고하셨습니당

frohsch committed:
[Fix] #72 - Modify network entity

frohsch committed:
[Fix] #72 - Modify layout

frohsch committed:
[Fix] #72 - minor fix
